systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,375 @@
+     *
+     * <p>Note: The {@code ResourceManager.setTestContext(currentContext);} is needed because this method is invoked in a new thread.
+     * Therefore, if you do not set the context, you would end up with a NullPointerException (NPE) because a new thread does not hold
+     * the state of the {@code ExtensionContext}, and so you need to set it.</p>
+     *
+     * @param namespace          The namespace name.
+     * @param clusterName        The cluster name.
+     * @param topicName          The topic name.
+     * @param currentContext     The current context of the test case.
+     */
+    public static void createTopicWithWait(String namespace, String clusterName, String topicName, ExtensionContext currentContext) {
+        ResourceManager.setTestContext(currentContext);
+        createTopic(namespace, clusterName, topicName, 12, 3, 2);
+        waitForTopicStatus(namespace, topicName, CustomResourceStatus.Ready, ConditionStatus.True);
+    }
+
+    /**
+     * Updates a KafkaTopic and waits until the configuration is updated.
+     *
+     * <p>Note: The {@code ResourceManager.setTestContext(currentContext);} is needed because this method is invoked in a new thread.
+     * Therefore, if you do not set the context, you would end up with a NullPointerException (NPE) because a new thread does not hold
+     * the state of the {@code ExtensionContext}, and so you need to set it.</p>
+     *
+     * @param namespace          The namespace name
+     * @param clusterName        The cluster name.
+     * @param topicName          The topic name.
+     * @param currentContext     The current context of the test case.
+     * @param configToUpdate     Configuration to update in the Kafka topics.
+     */
+    public static void updateTopicWithWait(String namespace, String clusterName, String topicName, ExtensionContext currentContext, Map<String, Object> configToUpdate) {
+        ResourceManager.setTestContext(currentContext);
+        updateTopic(namespace, topicName, new KafkaTopicSpecBuilder().withConfig(configToUpdate).build());
+        waitForTopicConfigContains(namespace, topicName, configToUpdate);
+    }
+
+    /**
+     * Deletes Kafka topics within a specified range and waits until they are fully deleted.
+     *
+     * <p>Note: The {@code ResourceManager.setTestContext(currentContext);} is needed because this method is invoked in a new thread.
+     * Therefore, if you do not set the context, you would end up with a NullPointerException (NPE) because a new thread does not hold
+     * the state of the {@code ExtensionContext}, and so you need to set it.</p>
+     *
+     * @param namespace          The namespace name.
+     * @param topicName          The topic name.
+     * @param currentContext     The current context of the test case.
+     */
+    public static void deleteTopicWithWait(String namespace, String topicName, ExtensionContext currentContext) {
```

Comment:
You don't have to pass the `ExtensionContext` anywhere or? You can do the `ResourceManager.getTestContext()`

Reply:
This is leftover (i.e., no longer used) cause I have to re-work it. Removing this code but thanks for pointing it out.